documentation/general/workloads/workload-history.md

```diff
@@ -0,0 +1,44 @@
+# Workload History
+
+When .NET SDK workloads are updated, it is possible that the new versions of the workloads won't work (if we shipped a bug) or won't be available (if the manifest updates are available but the packs themselves are not).  Right now it is hard to fix such a state.  We will add a `dotnet workload history` command that will show a history of the workload installation actions that have been run via the .NET CLI, and add new options on the `dotnet workload update` command that will allow rolling back to a previously installed version of workloads from the history.
+
+## Recording installation actions
+
+When a CLI command is run which modifies the installed workloads or updates the manifests, we will write a record to the workload history log.  The workload commands this applies to include `install`, `update`, `uninstall`, `repair`, and `restore`.  Each workload history record will include the following information:
+
+- Date/time command was run
```

Comment:
Do we really need all this information? I'm wondering why we need both time run and time completed, I'm wondering if we should be keeping failures _at all_ (given that they don't actually update workload state and we aren't rolling back to them), and I'm a bit iffy on the rollback file/workload state parts, not because those aren't reasonable things to include but because those are potentially _big_, and a file that's intended to persist and never go is a major problem if it grows that fast. We presumably aren't intending to delete it at any point (or what's the point of having the file in the first place?), so I can easily imagine this becoming a problem, especially if we really do include all this information. In particular, imagine if I specify the wrong file by accident as my rollback file. It's 10 GB. I include the whole thing in this file, but the command fails. I don't immediately realize what my mistake was, so I run it 2-3 more times, adding another 20-30 GB to the history file. Do we really want a 40 GB file here? And I would argue that if the user then deletes the file as "why do I have this massive thing in my SDK?", we've failed. I think it should ideally be as streamlined as possible to prevent users from ever hitting that problem unless they run 100 million commands.

Reply:
None of this should be big.  The workload state isn't supposed to include the actual contents of the workload packs or even the manifests.  "Installed workloads" just means the list of names of the workloads that are installed.  To roll back to a given state may still require downloading the workload packages from NuGet feeds, we aren't trying to avoid that.

Time run and time completed lets us know how long the command took to run, which could be useful.  I think we should try to gather all this information as well as other information that we think might be useful.  We'll be writing a separate data file for each command, and I would guess that each file would be 1 KB or less.

I think it's very unlikely that someone would accidentally specify a rollback file that was very large but still parsed correctly as a rollback file.  We don't need to include the rollback file contents verbatim, just what we parsed out of it successfully.  And if it fails to parse, then that's a failure that's early enough that I agree we wouldn't need to write a workload history record.  If we actually start to try downloading something or installing something, I think we should write the record even if the operation ends up failing.